### PR TITLE
feat: pimlico verifying paymaster

### PR DESCRIPTION
**Pull request by nksazonov**

### CODE REVIEW

**Review by philanton — APPROVED** [review]

File: pkg/userop/middleware.go [hunk 1]
```diff
@@ -283,6 +292,68 @@ func getPimlicoERC20PaymasterData(
 	}
 }
 
+func getPimlicoVerifyingPaymsterAndData(
```
Comment (maxpushka) [inline pkg/userop/middleware.go:295]: A little typo
```suggestion 
func getPimlicoVerifyingPaymasterAndData(
```

File: pkg/userop/middleware.go [hunk 2]
```diff
@@ -283,6 +292,68 @@ func getPimlicoERC20PaymasterData(
...
+		var gasEst gasEstimate
+		if pimlicoVerifyingConfig.SponsorshipPolicyID == "" {
+			if err := bundler.CallContext(ctx, &gasEst, "pm_sponsorUserOperation", opModified, entryPoint); err != nil {
+				return fmt.Errorf("failed to call pm_sponsorUserOperation: %w", err)
+			}
+		} else {
+			if err := bundler.CallContext(ctx, &gasEst, "pm_sponsorUserOperation", opModified, entryPoint, pimlicoVerifyingConfig); err != nil {
+				return fmt.Errorf("failed to call pm_sponsorUserOperation: %w", err)
+			}
+		}
```
Comment (maxpushka) [inline pkg/userop/middleware.go:340]: Just found a way to simplify branching:
```suggestion
    sponsorArgs := []any{opModified, entryPoint}
		if pimlicoVerifyingConfig.SponsorshipPolicyID != "" {
      sponsorArgs = append(sponsorArgs, pimlicoVerifyingConfig)
		}
    if err := bundler.CallContext(ctx, &gasEst, "pm_sponsorUserOperation", sponsorArgs...); err != nil {
      return fmt.Errorf("failed to call pm_sponsorUserOperation: %w", err)
    }
```

Comment (nksazonov) [reply pkg/userop/middleware.go:340]: Oh, nice. Thanks for suggestion!

File: pkg/userop/middleware.go (hunk 1 above)
Comment (nksazonov) [reply pkg/userop/middleware.go]: 3586234beace3d1ceef3bc83a46bd1ab554bfc48

**Review by maxpushka — APPROVED** [review]

**Pull request merged**